src/core/js/collections/notifyPushCollection.js

```diff
@@ -1,53 +1,49 @@
-define([
-  'core/js/adapt',
-  'core/js/views/notifyPushView',
-  'core/js/models/notifyModel'
-], function(Adapt, NotifyPushView, NotifyModel) {
-
-  // Build a collection to store push notifications
-  var NotifyPushCollection = Backbone.Collection.extend({
-
-    model: NotifyModel,
-
-    initialize: function() {
-      this.listenTo(this, 'add', this.onPushAdded);
-      this.listenTo(Adapt, 'notify:pushRemoved', this.onRemovePush);
-    },
-
-    onPushAdded: function(model) {
-      this.checkPushCanShow(model);
-    },
-
-    checkPushCanShow: function(model) {
-      if (this.canShowPush()) {
-        model.set('_isActive', true);
-        this.showPush(model);
-      }
-    },
-
-    canShowPush: function() {
-      var availablePushNotifications = this.where({ _isActive: true });
-      if (availablePushNotifications.length >= 2) {
-        return false;
-      }
-      return true;
-    },
-
-    showPush: function(model) {
-      new NotifyPushView({
-        model: model
-      });
-    },
-
-    onRemovePush: function(view) {
-      var inactivePushNotifications = this.where({ _isActive: false });
-      if (inactivePushNotifications.length > 0) {
-        this.checkPushCanShow(inactivePushNotifications[0]);
-      }
+import Adapt from 'core/js/adapt';
+import NotifyPushView from 'core/js/views/notifyPushView';
+import NotifyModel from 'core/js/models/notifyModel';
+
+// Build a collection to store push notifications
+export default class NotifyPushCollection extends Backbone.Collection {
+
+  model() {
+    return NotifyModel;
+  }
+
+  initialize() {
+    this.listenTo(this, 'add', this.onPushAdded);
+    this.listenTo(Adapt, 'notify:pushRemoved', this.onRemovePush);
+  }
+
+  onPushAdded(model) {
+    this.checkPushCanShow(model);
+  }
+
+  checkPushCanShow(model) {
+    if (this.canShowPush()) {
+      model.set('_isActive', true);
+      this.showPush(model);
     }
+  }
 
-  });
-
-  return NotifyPushCollection;
+  canShowPush() {
+    var availablePushNotifications = this.where({ _isActive: true });
+    if (availablePushNotifications.length >= 2) {
```

Comment:
could do a ternary statement here

It might also be worth declaring all variables with const or let depending on the type of variable used to make them block scoped?